### PR TITLE
Deprecate custom kernel

### PR DESCRIPTION
**Pull request by ltratt**
The Krun readme recommends using our crusty old custom Linux kernel, which probably isn't a good idea for most people. Rather than forcing nearly every Krus user to set `NO_MSRS=1`, this PR changes the flag to `MSRS` and defaults to `MSRS=0` (i.e. MSR support won't be build).

Warning: this is untested! @vext01 if you think this PR is not a terrible idea, might you be able to lightly test that Krun at least still builds and runs some basic task OK?

### CODE REVIEW

**Comment by vext01** [comment]
I was going to say: "if the tests pass, then we can probably roll with it", but the tests don't appear to be running.

I'll do some manual testing shortly.

**Comment by ltratt** [comment]
Thanks!

File: libkrun/libkruntime.c [hunk 1]
```diff
@@ -30,7 +30,7 @@
 #define ACTUAL_CLOCK_MONOTONIC    CLOCK_MONOTONIC
 #endif
 
-#if defined(__linux__) && !defined(NO_MSRS)
```
Comment (vext01) [inline libkrun/libkruntime.c:33]: Do I see it right that throughout the PR `NO_MSRS` was renamed to `MSRS`, but the surrounding logic wasn't negated? That has to be a mistake?

For example, in this case I think you mean:
```
#if defined(__linux__) && defined(MSRS)
```

and not:
```
#if defined(__linux__) && !defined(MSRS)
```

(remove the leading `!` negation after the `&&`)

Comment (ltratt) [reply libkrun/libkruntime.c:33]: Eek, you're quite right! Fixed in bf4daf9.

Comment (ltratt) [reply libkrun/libkruntime.c:33]: And as a bonus, make the comments correct too in 72cda92.

**Comment by vext01** [comment]
Using the following diff on top of your work, the branch builds and all tests pass (on bencher8):
```diff
diff --git a/libkrun/libkruntime.c b/libkrun/libkruntime.c
index 77c814d..fe4d56c 100644
--- a/libkrun/libkruntime.c
+++ b/libkrun/libkruntime.c
@@ -103,7 +103,7 @@ krun_mdata_bounds_check(int mdata_idx)
  */
 static uint64_t krun_pctr_val_mask = 0;
 
-#elif defined(__linux__) && defined(MSRS)
+#elif defined(__linux__) && !defined(MSRS)
 #elif defined(__OpenBSD__)
 // We do not yet support performance counters on OpenBSD
 #else
@@ -238,7 +238,7 @@ krun_get_fixed_pctr1_width()
 
     return fixed_ctr_width;
 }
-#elif defined(__linux__) && defined(MSRS)
+#elif defined(__linux__) && !defined(MSRS)
 #elif defined(__OpenBSD__)
 // We do not yet support performance counters on OpenBSD
 #else
@@ -272,7 +272,7 @@ krun_init(void)
     /* Reset all MSRs of interest */
     syscall(SYSCALL_KRUN_RESET_MSRS, krun_num_cores);
 
-#elif defined(__linux__) && defined(MSRS)
+#elif defined(__linux__) && !defined(MSRS)
 #elif defined(__OpenBSD__)
     // We do not yet support performance counters on OpenBSD
 #else
@@ -292,7 +292,7 @@ krun_done(void)
         free(krun_mdata[i].aperf);
         free(krun_mdata[i].mperf);
     }
-#elif defined(__linux__) && defined(MSRS)
+#elif defined(__linux__) && !defined(MSRS)
 #elif defined(__OpenBSD__)
     // We do not yet support performance counters on OpenBSD
 #else
@@ -360,7 +360,7 @@ krun_measure(int mdata_idx)
         fprintf(stderr, "krun_read_msrs() syscall failed\n");
         exit(EXIT_FAILURE);
     }
-#elif defined(__linux__) && defined(MSRS)
+#elif defined(__linux__) && !defined(MSRS)
     data->wallclock = krun_clock_gettime_monotonic();
 #elif defined(__OpenBSD__)
     data->wallclock = krun_clock_gettime_monotonic();
```

**Comment by ltratt** [comment]
0b54f08 incorporates your suggested change.

**Comment by vext01** [comment]
What do you want to do next? Strictly speaking we'd need another reviewer.

**Comment by ltratt** [comment]
Arguably you've just been very concrete in telling me where I was too stupid to get my first diffs right. I think it's OK for me to squash this down into a single commit and for you to merge it.

**Comment by vext01** [comment]
Please squash then :P

**Comment by ltratt** [comment]
Squashed.

**Comment by vext01** [comment]
Just realised that this requires manually merging.

**Pull request merged**